bool unfollow
bool is_full
